lib/app/AppSettings.js

```diff
@@ -39,13 +39,36 @@ class AppSettings {
     return this
   }
 
+  setAttachedExtensions (extensions) {
+    this.attachedExtensions = extensions
+    return this
+  }
+
+  /**
+   * @param {string} extension
+   * @returns {AppSettings}
+   */
+  attachExtension (extension) {
```

Comment:
should throw error if extension already attached

Reply:
extensions should be saved as an object, not just a single string

Reply:
should accept an array of extensions (attachExtensions, could also be an extra function that wraps this function, move code from action to this file)